compiler/src/iree/compiler/Codegen/Dialect/GPU/TargetUtils/ConfigUtils.h

```diff
@@ -26,6 +27,30 @@ LogicalResult setTileAndFuseLoweringConfig(IREE::GPU::TargetAttr target,
                                            mlir::FunctionOpInterface entryPoint,
                                            Operation *op);
 
+//===----------------------------------------------------------------------===//
+// Pass Pipeline Options
+//===----------------------------------------------------------------------===//
+
+using IREE::GPU::PipelineOptions::ReorderWorkgroupsStrategy;
+using IREE::GPU::PipelineOptions::UnitPipelineOption;
+
+StringRef getPipelineOptionName(UnitPipelineOption option);
+StringRef getPipelineOptionName(ReorderWorkgroupsStrategy option);
+
+struct GPUPipelineOptions {
+  bool enableReduceSharedMemoryBankConflicts = true;
+  bool prefetchSharedMemory = false;
+  bool enableUkernels = false;
+  std::optional<ReorderWorkgroupsStrategy> reorderStrategy;
```

Comment:
Thinking about it again, why do we need the optional if we can set this to None? Don't need to fix in this in this PR though.

Reply:
I think the uninitialized optional value will get replaced by a default value determined by a command line flag, so it makes a difference being optional vs None. It can probably be done better, but I will leave it alone for this PR.